controller/instance_manager_controller.go

```diff
@@ -751,15 +751,13 @@ func (imc *InstanceManagerController) enqueueKubernetesNode(obj interface{}) {
 func (imc *InstanceManagerController) cleanupInstanceManager(im *longhorn.InstanceManager) error {
 	log := getLoggerForInstanceManager(imc.logger, im)
 	im.Status.IP = ""
+	if len(im.Status.Instances) > 0 {
```

Comment:
There is no need to do any IM CR modfication if this function is just for deleting the im pod and stopping the im monitoring. (but we can clean up this later)